modules/nf-core/sentieon/haplotyper/main.nf

```diff
@@ -0,0 +1,85 @@
+process SENTIEON_HAPLOTYPER {
+    tag "$meta.id"
+    label 'process_medium'
+    label 'sentieon'
+
+    secret 'SENTIEON_LICENSE_BASE64'
+
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        exit 1, "Sentieon modules does not support Conda. Please use Docker / Singularity / Podman instead."
+    }
+
+    container 'docker.io/nfcore/sentieon:202112.06'
+
+    input:
+    val(emit_mode)
+    tuple val(meta), path(input), path(input_index), path(intervals)
+    path  fasta
+    path  fai
+    path  dbsnp
+    path  dbsnp_tbi
+
+    output:
+    tuple val(meta), path("*.unfiltered.vcf.gz")    , optional:true, emit: vcf   // added the substring ".unfiltered" in the filename of the vcf-files since without that the g.vcf.gz-files were ending up in the vcf-channel
+    tuple val(meta), path("*.unfiltered.vcf.gz.tbi"), optional:true, emit: vcf_tbi
+    tuple val(meta), path("*.g.vcf.gz")             , optional:true, emit: gvcf   // these output-files have to have the extension ".vcf.gz", otherwise the subsequent GATK-MergeVCFs will fail.
+    tuple val(meta), path("*.g.vcf.gz.tbi")         , optional:true, emit: gvcf_tbi
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''    // options for the driver
+    def args2 = task.ext.args2 ?: ''  // options for the vcf generation
+    def args3 = task.ext.args3 ?: ''  // options for the gvcf generation
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def dbsnp_command = dbsnp ? "-d $dbsnp " : ""
+    def interval_command = intervals ? "--interval $intervals" : ""
+    def sentieon_auth_mech_base64 = task.ext.sentieon_auth_mech_base64 ?: ''
+    def sentieon_auth_data_base64 = task.ext.sentieon_auth_data_base64 ?: ''
+    def vcf_cmd = ""
+    def gvcf_cmd = ""
+    def base_cmd = '--algo Haplotyper ' + dbsnp_command
+
+    if (emit_mode != 'gvcf') {
+        vcf_cmd = base_cmd + args2 + ' ' + prefix + '.unfiltered.vcf.gz'
+    }
+
+    if (emit_mode == 'gvcf' || emit_mode == 'both') {
```

Comment:
`emit_mode` can be `gvcf`, `both` or anything else (and that will be interpreted as `vcf`) 

https://github.com/nf-core/modules/blob/a746b933e61f43f8932aa2f867d5ec7f0ded352b/modules/nf-core/sentieon/haplotyper/meta.yml#L15-L21

Reply:
> Running it once would be a (a bit) more efficient, although Sentieon only passes over the BAM once unlike every other tool.

I'm not sure I understand what you mean. I set that up with the "chaining" of the vcf-command and the gvcf-command as recommended by Don.

Reply:
Yeah, it was also my first thought that from the gvcf you can get the vcf, but - as far as I remember - we weren't able to produce a vcf from the gvcf which was identical to the vcf we would get if just producing the vcf directly from the variant-caller. (Sorry, but I can't recall what the exact differences were; I could perhaps dig that out from old emails and Teams chats.) I believe Don or somebody from the team made me aware that Sentieon-haplotyper could just produce both a vcf and gvcf by "chaining" the commands, and so we went with that in DNGC's Sarek-Sentieon-pipeline, and it works fine there.

Of course, just because that setup worked for DNGC doesn't necessarily mean that it has to be done in the same way in the real Sarek pipeline.

My feeling is that the scenario, where one wants both a gvcf AND a vcf, is probably very rare, and so one could argue that it is simply not something that the real Sarek-pipeline is going to support. (It is also - to the best of my knowledge - not support by any of the other variant-caller-subworkflows in Sarek.)